exercises/linked-list/canonical-data.json

```diff
@@ -0,0 +1,543 @@
+          {
+            "operation": "count",
+            "expected": 2
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "8272cef5-130d-40ea-b7f6-5ffd0790d650",
+      "description": "count is correct after mutation",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 31
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "unshift",
+            "value": 43
+          },
+          {
+            "operation": "count",
+            "expected": 2
+          },
+          {
+            "operation": "shift"
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "count",
+            "expected": 0
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "229b8f7a-bd8a-4798-b64f-0dc0bb356d95",
+      "description": "popping to empty doesn't break the list",
+      "comments": [
+        "When not implemented correctly, head/tail reference will break when  ",
+        "the last item of the list is removed. This test adds an item after a ",
+        "list with at least two elements (different head than tail) has been  ",
+        "emptied out. If done correctly, the list will still work."
+      ],
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 41
+          },
+          {
+            "operation": "push",
+            "value": 59
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "push",
+            "value": 47
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "pop",
+            "value": 47
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "4e1948b4-514e-424b-a3cf-a1ebbfa2d1ad",
+      "description": "shifting to empty doesn't break the list",
+      "comments": [
+        "When not implemented correctly, head/tail reference will break when  ",
+        "the last item of the list is removed. This test adds an item after a ",
+        "list with at least two elements (different head than tail) has been  ",
+        "emptied out. If done correctly, the list will still work."
+      ],
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 41
+          },
+          {
+            "operation": "push",
+            "value": 59
+          },
+          {
+            "operation": "pop"
+          },
+          {
+            "operation": "pop"
```

Comment:
these are `pop`s, but the description says "shifting to empty doesn't break the list" - do I understand correctly to mean that these should be `shift` instead, or have I misunderstood?

Reply:
You have not. I think I broke it when I fixed something else. Let me fix this! Thanks for the fresh 👁️

Reply:
Hmm. It was actually the `pop` on line 347 that was incorrect, but fixed them all!